pkg/sign/signer.go

```diff
@@ -15,99 +15,240 @@
 package sign
 
 import (
-	"crypto"
-	"crypto/rand"
-	"crypto/sha256"
-	_ "crypto/sha512" // if user chooses SHA2-384 or SHA2-512 for hash
-	"crypto/x509"
+	"bytes"
 	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
 	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
 
 	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
 	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 )
 
+const bundleV03MediaType = "application/vnd.dev.sigstore.bundle.v0.3+json"
+
 type Signer interface {
-	Sign(data []byte) (*protobundle.Bundle, error)
+	Sign(content Content, k Keypair) (*protobundle.Bundle, error)
+}
+
+type Fulcio struct {
+	options *FulcioOptions
+}
+
+type FulcioOptions struct {
+	// URL of Fulcio instance
+	BaseURL string
+	// OIDC token to send to Fulcio instance
+	IdentityToken string
```

Comment:
Mentioned above, but I think this should be a part of each request rather than initializing the Fulcio struct, as this makes the Fulcio object harder to reuse.